concepts/solution-design-kafka-pull.md

```diff
@@ -0,0 +1,248 @@
+
+## Security and Token Management
+
+### Token Creation and Expiration
+
+- **Dynamic Token Issuance:**
+  The OAuth Service provides short-lived access tokens and refresh tokens using SASL/OAUTHBEARER. These tokens guarantee
+  that any consumer’s access is temporary and can be refreshed or revoked as needed.
+- **Secure Vault Storage:**
+  All dynamically issued credentials are saved within a secure Vault by the Kafka Extension. This allows both safe
+  retrieval during transfer start and secure cleanup upon transfer termination.
+
+### Token Refresh Strategies
+
+The design supports various token refresh strategies:
+
+- Automatic Refresh via Consumer DataPlane:
+  Proactively refresh tokens before they expire.
+- Automatic Refresh via the EDR API (/edrs):
+  The Provider Control Plane can handle token refresh operations when needed.
+- Manual Refresh:
+  The consumer application can trigger token refresh operations if required.
+
+These strategies ensure that the system can maintain uninterrupted data flow while complying with dynamic security
+policies.
+
+See the [Tractus-X EDC Signaling Extension](https://github.com/eclipse-tractusx/tractusx-edc/blob/main/docs/development/dataplane-signaling/tx-signaling.extensions.md)
+for more details.
+
+## Interoperability
+
+The Kafka Extension does not change anything related to IATP, DSP and policy definitions. This ensures full
+conformity to the
+Standard [CX-0018 Dataspace Connectivity v.3.1.0](https://catenax-ev.github.io/docs/standards/CX-0018-DataspaceConnectivity)
+for chapters 2.1, 2.3, 2.4 and 2.5.
+
+Since the Kafka PULL extension will introduce the new transfer type `Kafka-PULL` the standard has to be extended by this
+new type once the extension exits POC state and is introduced as a proper Tractus-X EDC extension.
+
+An example for the extended standard could be:
+
+> 2.2.3 Kafka-PULL
+> 
+> A Consumer MUST send a `dspace:TransferRequestMessage` with `dct:format:dspace:Kafka-PULL`.
+> 
+> A Provider MUST send a `dspace:TransferStartMessage` with sufficient information in the `dspace:dataAddress` property so that a client connection to the `dspace:endpoint` may succeed when initialized with the properties `scopes`, `groupPrefix` and `topic`.
+> 
+> A Provider Connector MUST ensure that the requested backend system has sufficient context from the negotiation to evaluate the legitimacy of the request.
+> 
+> A Consumer may then use the provided data to execute requests against the endpoint.
+> 
+> Despite the token, the endpoint still has the right to refuse serving a request. This may occur for instance when a consumer attempts to request for a different topic than the one specified in the `dspace:dataAddress`.
+
+## Enhancements Over the Existing POC
+
+The revised design improves upon the initial proof-of-concept by incorporating:
+
+- **Dynamic Credential Management:** Replacing hard-coded credentials with OAuth-driven dynamic client registration and token management.
```

Comment:
Is this a requirement and how to deal with that?

Reply:
For a secure data stream which allows revoking of access when a transfer process gets terminated, it is important to have short-lived access tokens. This requires to replace the hard-coded credentials, currently used in the POC, to be replaced by a token based authentication.